core/api/src/main/java/com/blazebit/persistence/Queryable.java

```diff
@@ -58,4 +59,11 @@
      */
     public T getSingleResult();
 
+    /**
+     * Execute the query and return the result as a type Stream.
+     *
+     * @return The stream of the results
+     */
+    public Stream<T> getResultStream();
```

Comment:
Quickstart still works under Java 7, although this change leads to a `ClassNotFoundException` when doing reflection calls on the Criteria Builder API, i.e.:

```java
for (Method method : cb.getClass().getMethods()) {
    System.out.println(method);
}
```